types/oscal/oscal.go

```diff
@@ -183,30 +148,34 @@ func New(r io.Reader) (*OSCAL, error) {
 				return &OSCAL{Profile: &profile}, nil
 			}
 		}
-
-		return nil, errors.New("Malformed OSCAL")
 	}
 
 	return nil, errors.New("Malformed OSCAL. Must be XML or JSON")
 }
 
-// RawXML ...
-func (o *OSCAL) RawXML(prettify bool) ([]byte, error) {
+// XML writes the OSCAL object as XML to the given writer
+func (o *OSCAL) XML(w io.Writer, prettify bool) error {
+	e := xml.NewEncoder(w)
 	if prettify {
-		return xml.MarshalIndent(o, "", "  ")
+		e.Indent("", "  ")
+		return e.Encode(o)
 	}
-	return xml.Marshal(o)
+
+	return e.Encode(o)
 }
 
-// RawJSON ...
-func (o *OSCAL) RawJSON(prettify bool) ([]byte, error) {
+// JSON writes the OSCAL object as JSON to the given writer
+func (o *OSCAL) JSON(w io.Writer, prettify bool) error {
+	e := json.NewEncoder(w)
 	if prettify {
-		return json.MarshalIndent(o, "", "  ")
+		e.SetIndent("", "  ")
+		return e.Encode(o)
```

Comment:
Same change suggested as the one on line 161. 

Also, I see a lot of common code in this method and the xml one apart from the statement where an encoder is created. Do you think we can extract a new method and use that method here and above?

Reply:
Good catch, thanks.